conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala

```diff
@@ -85,8 +90,12 @@ class TezosIndexer private (
   ): Unit = {
     val noOp = Future.successful(())
     val processing = for {
-      unhandled <- accountsResetHandler.applyUnhandledAccountsResets(accountResetEvents)
-      _ <- processTezosBlocks()
+      maxLevel <- indexedData.fetchMaxLevel
+      reloadedAccountEvents <- processFork(maxLevel)
```

Comment:
This opens a small can of worms, just gimme time to consider throughly and I'll get back

Reply:
The crux is that the detector decides if a fork happened based on the current tip of the local indexed chain.
This means there's a small hole in the logic, assuming the following scenario:

- the detector checks the tip and it matches the node block at that level
- the indexer fetches a few blocks (suppose 3) but during the fetch the fork happens and the last block happens to be on the fork
- since we don't check for consistency of fetched blocks (via predecessor), it's all fine for Lorre
- at next cycle, the detector will check the tip and find that it still matches correctly the node block at that level and won't spot the error

This leads to a situation where the first 2 blocks of the fetch cycle are now invalid but undetected.

Reply:
I guess the answer is related to your proposal:
 - to store the latest level that the detector checked in previous cycles and when the next detection check is done, verify all the blocks in between...

This complicates a bit this phase and carries additional risks to be handled.

@vishakh: I'd like your opinion on how to best proceed. I'd suggest going on with this (strictly speaking: incorrect) solution, and add the corrected checking as part of the next train of changes related to handling missing cases of invalidation (i.e. big-maps handling).
But you might want to wait and handle this first and now.

Reply:
> I'd suggest going on with this (strictly speaking: incorrect) solution, and add the corrected checking as part of the next train of changes related to handling missing cases of invalidation (i.e. big-maps handling).

If you mean let's go with what we have at the moment and fix this edge case in a future PR, then I fully agree. This problem requires some out of the box thinking before we work on an implementation.

Reply:
Opened a dedicated issue